### ISSUE TITLE
Allow custom usage of storage

### ISSUE THREAD

**Issue by leplatrem** [mozilla-services/cliquet#243]
Currently storage methods take a ``cliquet.Resource`` object as first parameter and are not very low level.

I suggest that we refactor/change this to allow low-level create/update/delete of ``records(str, str, json)``.

```python
request.storage.create('__history__', '__internal__', {"action": "click"})
```

```python
request.storage.create('__schema:' + collection, userid, {"title": "form"})
```

We have several possibilities:

1. Rename current methods with a ``resource_`` prefix (or ``_record`` suffix, whatever) and add methods for low-level (get, create, update, delete).
1. Rewrite current backends to perform only low-level operations, and wrap them to provide a resource storage backend like the current one
1. Add another group of backends implementation (e.g. leveldb) (in addition to cache and storage) to provide low-level storage (with a schema similar to cache)
1. Got to go to eat right now.

**Comment by ametaireau** [comment]
Hey, and we'll need this for schema storage!

Having the storage expose two different set of APIs would work, I think. For instance, in order to store records you would do `storage.collection.create(blah)` (as it is now, without the collection part) and have another `storage.raw.create()` api which is used by the first one.

**Comment by leplatrem** [comment]
I would like to avoid duplication of raw/collection logics for each storage backend...  I cannot think of a very nice way to do this :(

We are not completely stuck for Kinto though:

```python
class SchemaResource(BaseResource):
    @property
    def name(self):
        return '__schema_%s' % super(SchemaResource, self).name

    # ...  KintoResource ...
    def store_schema(self, schema):
        resource = SchemaResource(self.request)
        self.storage.create(resource, self.user_id, schema)
```

**Comment by ametaireau** [comment]
Yep that's a hack, but that would work ;)

**Comment by leplatrem** [comment]
Oh, I read it again, and actually it won't because ``SchemaResource.name`` will always be the same.

It would better be: 

```python
class KintoResource(BaseResource):
    # ...
    # ...
    def store_schema(self, schema):
        resource = BaseResource(self.request)
        resource.name = '__schema_%s' % self.name
        self.storage.create(resource, self.user_id, schema)
```
I'll add a setter to ``resource.name`` then.

**Comment by leplatrem** [comment]
@Natim could you go this way for Kinto ?

**Comment by leplatrem** [comment]
Another way of doing it is to use the *cache* backend.

That implies a little bit of work regarding wording.

Currently: 
* **cliquet.cache** : key-value + TTL
* **cliquet.storage**: records of resources

It could be:
* **cliquet.backends.cache**: key-value + TTL
* **cliquet.backends.keyvalue**: key-value
* **cliquet.backends.resource**: records of resources
* **cliquet.backends.attachments**: attachments of records

**Comment by leplatrem** [comment]
Implemented in #282
